components/appmgt-extensions/org.wso2.carbon.appmgt.mdm.wso2mdm.direct/src/main/java/org/wso2/carbon/appmgt/mdm/wso2mdm/direct/MDMOperationsImpl.java

```diff
@@ -0,0 +1,308 @@
+
+    private static final String APPM_APP_TYPE = "type";
+    private static final String APPM_APP_NAME = "name";
+    private static final String APPM_APP_IDENTIFIER = "identifier";
+    private static final String APPM_APP_IMAGE_URL = "ImageURL";
+    private static final String APPM_APP_TYPE_ENTERPRISE = "enterprise";
+    private static final String APPM_APP_TYPE_PUBLIC = "public";
+    private static final String APPM_APP_TYPE_WEBAPP = "webapp";
+    private static final String APPM_APP_TYPE_MOBILE_DEVICE = "mobileDevice";
+    private static final String APPM_APP_PLATFORM = "platform";
+    private static final String APPM_APP_PLATFORM_IOS = "ios";
+    private static final String APPM_APP_PLATFORM_WEBAPP = "webapp";
+    private static final String APPM_DEVICE_ID_CONCAT_SEPARATOR = "---";
+    private static final String APPM_APP_CALL_TYPE_USER = "user";
+    private static final String APPM_APP_CALL_TYPE_ROLE = "role";
+    private static final String APPM_APP_CALL_TYPE_STATUS = "status";
+
+    @Override
+    public void performAction(User currentUser, String action, App app, int tenantId, String type, String[] params,
+            HashMap<String, String> configParams) {
+        try {
+            //            DeviceManagementProviderService deviceManagementProviderService = getDeviceManagerService(
+            //                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+
+            ApplicationManager appManagerConnector = getAppManagementService(
+                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            MobileAppBean mobileApp = createMobileApp(currentUser, action, app, tenantId, type, params, configParams);
+            List<DeviceIdentifier> deviceIdentifiers = convertDeviceIdentifiers(type, params);
+            for (DeviceIdentifier deviceIdentifier : deviceIdentifiers) {
+                org.wso2.carbon.device.mgt.common.operation.mgt.Operation operation = null;
+                ArrayList<DeviceIdentifier> deviceIdentifiersToInstall = new ArrayList<>(1);
+                if (deviceIdentifier.getType().equals(Platform.android.toString())) {
+                    operation = MDMAndroidOperationUtil.createInstallAppOperation(mobileApp);
+                } else if (deviceIdentifier.getType().equals(Platform.ios.toString())) {
+                    operation = MDMIOSOperationUtil.createInstallAppOperation(mobileApp);
+                } else {
+                    log.error("Could not infer the operation on mobile device. User [" + currentUser + "] Application ["
+                            + mobileApp + "]");
+                }
+                if (operation != null) {
+                    appManagerConnector.installApplication(operation, deviceIdentifiersToInstall);
+                }
+            }
+
+        } catch (AppManagementException | ApplicationManagementException e) {
+            log.error("Could not retrieve the 'DeviceManagementProviderService' by OSGI lookup. User [" + currentUser
```

Comment:
The error message logged doesn't have any connection with "ApplicationManagementException". In other words, the aforementioned exception is thrown only if there's some issue in installing an application via the appManagerConnector. Will have to be fixed.